add fixed type support

muradm opened the pull request:
Fixed type allows to carry byte arrays of predefined size. Very useful for some types.

Fristi commented on core/src/main/scala/formulation/AvroDecoder.scala:
```diff
@@ -113,6 +113,9 @@ object AvroDecoder {
     override val instant: AvroDecoder[Instant] = long.andThen(ts => Either.fromTry(Try(Instant.ofEpochMilli(ts))))
 
+    override def fixed(name: String, size: Int, namespace: Option[String] = None): AvroDecoder[Array[Byte]] =
+      partial { case (_, s, v: GenericFixed) => Validated.valid(v.bytes()) }
```
I havent't looked at the API yet, but should you assert `v` is a certain size? And is `v.bytes()` side-effectful?

muradm replied:
i'm not clearly understanding the library yet. this pattern match, where the values come from? from record being decoded, right? if so, then those values received from Avro, if received, then they are valid, isn't it? what is the point of validation?

Fristi commented on core/src/main/scala/formulation/AvroDefaultValuePrinter.scala:
```diff
@@ -43,6 +43,9 @@ object AvroDefaultValuePrinter {
     }
 
+    override def fixed(name: String, size: Int, namespace: Option[String] = None): AvroDefaultValuePrinter[Array[Byte]] =
+      by(byteArray) { bytes => bytes }
```
This could be just `byteArray`

Fristi commented:
StringSchemaTest test is failing, but that's happening from time to time. I think we could fix this by using the latest scalacheck version (which prints a hash of the variables being used to reproduce the issue).

muradm commented:
added encoding/decoding tests for `fixed`, fixed printer and a minor warning.

comments for validation still valid :) yeah, we could repeat Avro's job, but I don't see point of validating decoded messages. If Avro did't, then user has to bear with it.
